### PR TITLE
Winhttp - Adding option to ignore client certificate request from service

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/3548

Enhances: https://github.com/Azure/azure-sdk-for-cpp/pull/3549

### CODE REVIEW

File: doc/HttpTransportAdapter.md [hunk 1]
```diff
@@ -221,8 +221,13 @@ The second step is to implement the method shown below in the global unnamed nam
 */
 
 // From one of the cpp files in your application:
-std::shared_ptr<Azure::Core::Http::HttpTransport> ::AzureSdkGetCustomHttpTransport() {
+std::shared_ptr<Azure::Core::Http::HttpTransport> ::AzureSdkGetCustomHttpTransport(
+  Azure::Core::Http::HttpServiceTransportOptions options) {
```
Comment (gearama) [inline doc/HttpTransportAdapter.md:225]: should we have a default for options ? that way it wouldn't be breaking change

**Review by gearama — APPROVED** [review]

**Comment by vhvb1989** [comment]
Closing in favor of: https://github.com/Azure/azure-sdk-for-cpp/pull/3585

**Pull request closed without merging**